### PR TITLE
ENH: Read from postgis in chunks

### PR DESCRIPTION
**Pull request by schilli**
## Motivation
Pandas allows to read rows from a database connection in chunks with the `chunksize` parameter to the functions `read_sql()`, `read_sql_table()` and `read_sql_query()`.
This is useful for parallelization of data processing in producer/consumer schemes, e.g. where a queue should be filled with chunks of data.
Under the hood this is handled by sqlalchemy's `fetchmany(chunksize)`.
It would be very useful to have this functionality in geopandas as well.

### `geopandas/io/sql.py`
* Moved all DataFrame postprocessing and the conversion to a GeodDataFrame from `read_postgis()` to the new function `_df_to_geodf()`
* Analogous to the implementaion of pandas' `read_sql()` functions,
    * the complete data is read into a single dataframe and then transformed into a geodataframe in the absence of `chunksize`, as before.
    * If `chunksize` is set, `pd.read_sql()` returns a generator that produces DataFrames. Then another generator, adding _df_to_geodf() is wrapped around it to yield the GeoDataFrames in chunks.
* The new `chunksize` parameters has been documented in the Docstrings.

### `geopandas/geodataframe.py`
* The new `chunksize` parameter was also added to `GeoDataFrame.from_postgis()` for consistent behaviour (where it is just passed to `read_postgis()`).
* The new `chunksize` parameters has been documented in the Docstrings.

### Tests
Tests have been implemented for the new `chunksize` parameter for `GeoDataFrame.from_postgis()` and `io.sql.read_postgis()`. `chunksize` has been set to 10, which is hopefully reasonable for the real tests.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
This seems a reasonable enhancement to me. Thanks for that.

We did some big changes in master in the meantime (started using the [black](https://github.com/psf/black) formatting, see https://github.com/geopandas/geopandas/blob/master/CONTRIBUTING.md#style how to deal with that). Are you OK with updating this on master? If you run into problems, I am happy to do it as well (the reformatting on master might given some annoying merge conflicts)

File: geopandas/io/sql.py [hunk 1]
```diff
@@ -7,60 +7,27 @@
...
-    coerce_float=True,
-    parse_dates=None,
-    params=None,
-):
+def _df_to_geodf(df, geom_col="geom", crs=None):
     """
-    Returns a GeoDataFrame corresponding to the result of the query
-    string, which must contain a geometry column in WKB representation.
+    Transfoms a pandas DataFrame into a GeoDataFrame.
+    The column geo_col must be a geometry column in WKB representation.
```
Comment (jdmcbr) [inline geopandas/io/sql.py]: A little more description about this being intended for dataframes generated from pandas sql reads of geospatial databases would probably be helpful. Also, "transfoms" -> "transforms" and "geo_col" -> "geom_col"

**Comment by jdmcbr** [comment]
@schilli Thanks for the PR! I resolved the conflicts in merging master, would probably be good to give my conflict resolution a once over just in case.

**Comment by martinfleis** [comment]
This looks ready to be merged, but is currently not tested (#1389).

File: geopandas/io/tests/test_sql.py [hunk 2]
```diff
@@ -292,6 +294,20 @@ def test_read_postgis_binary(self, connection_spatialite, df_nybb):
         df = read_postgis(sql, con, geom_col=geom_col)
         validate_boro_df(df)
 
+    def test_read_postgis_chunksize(self, connection_postgis, df_nybb):
+        """Test chunksize argument"""
+        chunksize = 10
```
Comment (jorisvandenbossche) [inline geopandas/io/tests/test_sql.py:299]:
```suggestion
        chunksize = 2
```

(since nybb has only 5 rows)

**Comment by jorisvandenbossche** [comment]
@schilli Thanks! (and sorry for the slow follow-up ;))

**Pull request merged**